In [1]:
import os
import sys
import numpy as np
import pylab as pl
import json
import os

In [2]:
alpha=0.05
# we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0 = 3.5 * 0.01 
P_1 = 70.1 * 0.01

if P_0 - P_1 >= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")

n_0 = 409
n_1 = 564

#lets get the counts by multiplying by the sample size
Nt_0 = P_0 * n_0
Nt_1 = P_1 * n_1

we must assess the statistical significance


In [3]:
#define the sample proportion first
sp = (P_0 * n_0 + P_1 * n_1) / (n_1 + n_0)
print (sp)

0.421047276465


In [4]:
# I am goonna create a little one line function to calculate the standard error
# and to calculate p
p = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)
#standard error
se = lambda p, n0, n1: np.sqrt(p * (1 - p) * (1.0 / n0 + 1.0 / n1))

In [5]:
zscore = lambda p0, p1, s : (p0 - p1) / s
z_2y = zscore(P_1, P_0, se(p(P_0, P_1, n_0, n_1), n_0, n_1))
print (z_2y)

20.7697865408


In [6]:
p_2y = 1 - 0.9984


def report_result(p,a):
    print ('is the p value ' + 
           '{0:.2f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('the Null hypothesis is {}'.format(\
                            'rejected' if p < a  else 'not rejected') )

    
report_result(p_2y, alpha)

is the p value 0.00 smaller than the critical value 0.05?
YES!
the Null hypothesis is rejected


### Null Hypothesis rejected:
Given the Z value of 20.76 we can say that for a significance level of 0.05 we reject the Null Hypothesis


# For new parameters:

Null hypothesis: the percentage of prisoners convicted of a felony that participated in the program is equal or higher than the percentage of prisoners convicted of a felony that didn't participated in the program.

Alternative hypothesis: the percentage of prisoners convicted of a felony that participated in the program is lower than the percentage of prisoners convicted of a felony that didn't participated in the program.


In [7]:


p_1 = 10.0 * .01
p_0 = 11.7 * .01

if p_0 - p_1 <= 0:
    # we are done
    print ("the Null holds")
else:
    print ("we must assess the statistical significance")
    
n_0 = 409
n_1 = 568

Nt_0 = p_0 * n_0
Nt_1 = p_1 * n_1



we must assess the statistical significance


In [8]:
sp = (p_0 * n_0 + p_1 * n_1) / (n_1 + n_0)
print (sp)

0.107116683726


In [9]:
z = zscore(p_1, p_0, se(p(p_0, p_1, n_0, n_1), n_0, n_1))
print(z)

-0.84764020522


In [22]:
p_val = 1 - 0.7995
report_result(p_val, alpha)

is the p value 0.20 smaller than the critical value 0.05?
NO!
the Null hypothesis is not rejected


In [23]:
print(p_val)

0.2005


### With the new parameters we cannot reject the null hypothesis

# Part 2 chi square

In [27]:
def evalChisq(values):
    '''Evaluates the chi sq from a contingency value
    Arguments:
    values: 2x2 array or list, the contingengy table
    '''
    if not (len(values.shape) == 2 and values.shape == (2,2)):
        print ("must pass a 2D array")
        return -1
    values = np.array(values)
    E = np.empty_like(values)
    for j in range(len(values[0])):
        for i in range(2):
            
            E[i][j] = ((values[i,:].sum() * values[:,j].sum()) / 
                        (values).sum())
    return ((values - E)**2 / E).sum()

In [28]:
Ntot = 977 # a + b + c + d = tot

sample_values = np.array([[0.1 * 568, 0.9 * 568], [0.117 * 409, 409 * 0.883]])
sample_values

array([[  56.8  ,  511.2  ],
       [  47.853,  361.147]])

In [30]:
print evalChisq(sample_values)

0.718493917505



# Chi square answers

0.71 is smaller than 3.84

why am I mentioning 3.84?

#### because that's where you reject with 95 % confidence! (check table for 1 degree of freedom).

how does the chi square statistic that you derived compare?

#### It is not enough to reject the null hypothesis. So you do not reject it.

please state what that means in terms of your Null hypothesis in a markdown cell below!
!!!!!! TODO FOR YOU!

lets see what the chi sq statistics says about the conviction for fellonies (row 10)

    fill in the contingency table below *

|** convicted of a fellony| 	yes| 	no**|
|:----------:|:----------|:------------|
|test sample|56.8|511.2| 			
|control sample|47.853|361.147|  			
|total|104.6|872.3|

# What not rejecting means.

Given that we did not reject the Null Hypothesis it is not possible to say with a 95% confidence level that prisoners that participated in the program have less probability of being convicted for a felony